pkg/util/validation/validation.go

```diff
@@ -118,15 +118,22 @@ func ValidateSpreadConstraint(spreadConstraints []policyv1alpha1.SpreadConstrain
 		}
 
 		if len(constraint.SpreadByField) > 0 {
-			spreadByFields.Insert(string(constraint.SpreadByField))
+			marked := spreadByFieldsWithErrorMark[constraint.SpreadByField]
```

Comment:
How about writing the code like this below?
```go
	spreadByFieldsWithErrorMark := make(map[policyv1alpha1.SpreadFieldValue]struct{})
        ...
	if len(constraint.SpreadByField) > 0 {
		_, marked := spreadByFieldsWithErrorMark[constraint.SpreadByField]
		if marked {
			allErrs = append(allErrs, field.Invalid(fldPath, spreadConstraints, fmt.Sprintf("multiple %s spread constraints are not allowed", constraint.SpreadByField)))
			continue
		}
		spreadByFieldsWithErrorMark[constraint.SpreadByField] = struct{}{}
	}
```

Reply:
There will be nine duplicated region errors if ten region spread constraints exist.